shared/macros/10-ansible.jinja

```diff
@@ -705,6 +705,59 @@ The following macro remediates Audit syscall rule in :code:`/etc/audit/audit.rul
 '"{{{ package }}}" in ansible_facts.packages'
 {{%- endmacro -%}}
 
+
+{{#
+    Macro used to check if authselect files are intact. When used, it will exit the respective
+    script if any authselect file was modified without proper use of authselect tool and
+    respective profiles.
+#}}
+{{% macro ansible_check_authselect_integrity() -%}}
+- name: '{{{ rule_title }}} - Check integrity of authselect current profile'
+  ansible.builtin.command:
+    cmd: authselect check
+  register: result_authselect_check_cmd
+  changed_when: false
+  ignore_errors: yes
+
+- name: '{{{ rule_title }}} - Informative message based on the authselect integrity check result'
+  ansible.builtin.assert:
+    that:
+      - result_authselect_check_cmd is success
+    fail_msg:
+      - authselect integrity check failed. Remediation aborted!
+      - This remediation could not be applied because an authselect profile was not selected or the selected profile is not intact.
+      - It is not recommended to manually edit the PAM files when authselect tool is available.
+      - In cases where the default authselect profile does not cover a specific demand, a custom authselect profile is recommended.
+    success_msg:
+      - authselect integrity check passed
```

Comment:
Not necessary for bash. It wouldn't be mandatory for Ansible too. However, since this resource is available in the `assert` module, I considered good to show this in the playbook results.